specifyweb/frontend/js_src/lib/components/Molecules/DateElement.tsx

```diff
@@ -19,14 +19,42 @@ export function DateElement({
   readonly fallback?: React.ReactNode;
   readonly flipDates?: boolean;
 }): JSX.Element {
+  const [relativeDate, setRelativeDate] = React.useState<string | undefined>(
+    date ? getRelativeDate(new Date(date)) : undefined
```

Comment:
this is a possible bug:
any time the default value of your state depends on a prop, you are likely creating a bug
this is because the default value is only used first time the component is rendered - so if the `date` prop would change later, the code will be out of date.

---

edit:
in this case, the bug is fixed thanks to useEffect having `date` as a dependency

but in the future, whenever your default value depends on a prop, instead of `React.useState`, use `useTriggerState` or `useLiveState`